ledger-service/fetch-contracts/src/main/scala/fetchcontracts/AcsTxStreams.scala

```diff
@@ -93,29 +94,26 @@ private[daml] object AcsTxStreams {
   private[daml] def transactionsFollowingBoundary(
       transactionsSince: lav1.ledger_offset.LedgerOffset => Source[Transaction, NotUsed]
   ): Graph[FanOutShape2[
-    BeginBookmark[String],
+    BeginBookmark[domain.Offset],
     ContractStreamStep.Txn.LAV1,
-    BeginBookmark[String],
+    BeginBookmark[domain.Offset],
   ], NotUsed] =
     GraphDSL.create() { implicit b =>
       import GraphDSL.Implicits._
-      type Off = BeginBookmark[String]
+      type Off = BeginBookmark[domain.Offset]
       val dupOff = b add Broadcast[Off](2)
       val mergeOff = b add Concat[Off](2)
       val txns = Flow[Off]
-        .flatMapConcat(off => transactionsSince(domain.Offset.tag.subst(off).toLedgerApi))
+        .flatMapConcat(off => transactionsSince(off.toLedgerApi))
         .map(transactionToInsertsAndDeletes)
       val txnSplit = b add project2[ContractStreamStep.Txn.LAV1, domain.Offset]
       import domain.Offset.`Offset ordering`
       val lastTxOff = b add last(LedgerBegin: Off)
-      type EndoBookmarkFlow[A] = Flow[BeginBookmark[A], BeginBookmark[A], NotUsed]
-      val maxOff = b add domain.Offset.tag.unsubst[EndoBookmarkFlow, String](
-        max(LedgerBegin: BeginBookmark[domain.Offset])
-      )
+      val maxOff = b add max(LedgerBegin: Off)
       // format: off
       discard { txnSplit.in <~ txns <~ dupOff }
       discard {                        dupOff                                       ~> mergeOff ~> maxOff }
```

Comment:
Whitespace fixing for the change on the next line. (Hard to see the difference in this comment; just align the `mergeOff`s.)

```suggestion
      discard {                        dupOff                                ~> mergeOff ~> maxOff }
```